src/library/dlgcoverartfullsize.cpp

```diff
@@ -131,8 +132,9 @@ void DlgCoverArtFullSize::slotCoverFound(const QObject* pRequestor,
             dialogSize.scale(availableScreenSpace.width(), dialogSize.height(),
                              Qt::KeepAspectRatio);
         }
-        QPixmap resizedPixmap = m_pixmap.scaled(size(),
```

Comment:
The issue is in line 125 above, where the logical dialogSize is taken form the native pixel size. 
Can you try to use 
QSize dialogSize = m_pixmap.size() / getDevicePixelRatioF(this);
there?